### PR TITLE
[keyboard] Add pico as a new keyboard

### PR DESCRIPTION
**Pull request by x1-**
## Description

Add `pico` as a new keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/pico/rules.mk [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight.
```
Comment (noroadsleft) [inline keyboards/pico/rules.mk]: Only change I'd like to see is removal of the firmware size impact numbers from Lines 17-20 and 27. Those numbers are rarely if ever accurate, so we're trying to get rid of them.

Everything else looks good to me.

Comment (x1-) [reply keyboards/pico/rules.mk]: @noroadsleft Thank you for reviwing!
I agree with you, so removed those lines. (https://github.com/qmk/qmk_firmware/pull/7419/commits/d0ff2358dcbe590b7d335dfd3ae63248972a5fb6)

File: keyboards/pico/rules.mk [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+COMMAND_ENABLE = yes        # Commands for debug and configuration
```
Comment (fauxpark) [inline keyboards/pico/rules.mk]:
```suggestion
BOOTMAGIC_ENABLE = no      # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
```

You removed the lines entirely, noroads just meant the part in `()` :)

File: keyboards/pico/rules.mk [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
```
Comment (fauxpark) [inline keyboards/pico/rules.mk]:
```suggestion
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
MIDI_ENABLE = no            # MIDI support
```

**Comment by x1-** [comment]
@fauxpark Thank you for suggestion!
I modified `rules.mk`!

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**